mos_tests/failover/python_tests/test_restarts.py

```diff
@@ -0,0 +1,200 @@
+                    results.append(not result['exit_code'])
+            # Will return True only if the cmd above
+            # returned zero exit code for each controller
+            return all(results)
+
+        logger.info('wait until all rabbits come alive')
+        wait(is_rabbit_alive, timeout_seconds=10 * 60, sleep_seconds=5)
+
+        logger.info('check that all rabbitmq nodes are in the same cluster')
+        # OSTF contain test "RabbitMQ availability"
+        # which will check if all rabbit nodes are up or not.
+        # Also it was found that restart of rabbit service
+        # affects other resources like p_nova_compute_ironic.
+        # And full recovery time might take from several up to 15 minutes.
+        # So the below call seems to be the best check in this case.
+        # It will guaranty that all services are recovered.
+        self.env.wait_for_ostf_pass()
+
+        self.check_common_services()
+
+    def test_instance_folder_after_hard_reboot(self, clean_os):
+        """ Re-creating instance folder after hard reboot
+
+        Scenario:
+            1. Deploy cloud with at least 1 compute node and 3 controllers
+            2. Create VM on each compute
+            3. Login to each compute node and shutdown it:
+                "shutdown now -r"
+            4. Perform "hard reboot" for all VMs
+            5. Hard reboot should help to recover all VMs
+               and they should be in ACTIVE state after that
+        """
+
+        # init variables
+        zone = self.os_conn.nova.availability_zones.find(zoneName="nova")
+        hosts = zone.hosts.keys()
+        security_group = self.os_conn.create_sec_group_for_ssh()
+        self.instance_keypair = self.os_conn.create_key(key_name='instancekey')
+
+        # create router
+        router = self.os_conn.create_router(name="router01")['router']
+        logger.info('router {} was created'.format(router['id']))
+
+        # create networks by amount of the compute hosts
+        for hostname in hosts:
+            net_id = self.os_conn.add_net(router['id'])
+            self.os_conn.add_server(net_id,
+                                    self.instance_keypair.name,
+                                    hostname,
+                                    security_group.id)
+
+        logger.info('shutdown all computes in the cluster')
+        computes = self.env.get_nodes_by_role('compute')
+        for compute in computes:
+            with compute.ssh() as remote:
+                cmd = 'shutdown now -r'
+                remote.check_call(cmd)
+
+        logger.info('Execute hard reboot for the affected servers')
+        # That should start back the compute nodes and enable the servers
+        for server in self.os_conn.nova.servers.list():
+            server.reboot(reboot_type='HARD')
+
+        logger.info('wait until all servers come in ACTIVE state')
+        for server in self.os_conn.nova.servers.list():
+            def is_server_active():
```

Comment:
let's create method in os_actions (same code here https://github.com/Mirantis/mos-integration-tests/blob/master/mos_tests/environment/os_actions.py#L159)